spec/integration/apply_spec.rb

```diff
@@ -4,266 +4,288 @@
 require 'bolt_spec/conn'
 require 'bolt_spec/files'
 require 'bolt_spec/integration'
+require 'bolt_spec/project'
 require 'bolt_spec/puppet_agent'
 require 'bolt_spec/run'
 
-describe "apply", expensive: true do
+TEST_VERSIONS = [
+  [5, 'puppet5'],
+  [6, 'puppet6'],
+  [7, 'puppet']
+].freeze
+
+describe 'apply', expensive: true do
   include BoltSpec::Conn
-  include BoltSpec::Files
   include BoltSpec::Integration
+  include BoltSpec::Project
   include BoltSpec::PuppetAgent
   include BoltSpec::Run
 
-  let(:modulepath) { File.join(__dir__, '../fixtures/apply') }
-  let(:hiera_config) { File.join(__dir__, '../fixtures/configs/empty.yml') }
-  let(:config_flags) { %W[--format json --targets #{uri} --password #{password} --modulepath #{modulepath}] + tflags }
-  let(:apply_settings) { { 'show_diff' => true } }
+  let(:project_config) do
```

Comment:
We should pass this transparently to `with_project` so it's not appearing magically.